lib/SelectedObjectsProvider/useObjectSelector.js

```diff
@@ -0,0 +1,49 @@
+import { useContext } from 'react';
+import { SelectedObjectsContext } from './index';
+
+export function useObjectSelector(
+  id,
+  type,
+  idsToSelect,
+  disabledCallback = () => false
+) {
+  const {
+    selected,
+    selectMultiple,
+    deselectMultiple,
+    updateSelected,
+    currentSelectedType,
+    intendedToSelect,
+    setIntendedToSelect
+  } = useContext(SelectedObjectsContext);
+
+  const isSelected = selected.indexOf(id) !== -1;
+  const isDisabled = disabledCallback(currentSelectedType, isSelected);
+  const isHovered = intendedToSelect.indexOf(id) !== -1;
+
+  const getIds = (additionalIds = []) => [
+    ...new Set([id, ...idsToSelect, ...additionalIds])
```

Comment:
> We wanted to seperate the first argument, id, with the ones being selected, idsToSelect. This is so we're certain that it will only selected the id's that are in idsToSelect and nothing more.

This is a cool idea that gives you a nice bit of flexibility!

Should this line therefore be?
```suggestion
    ...new Set([...idsToSelect, ...additionalIds])
```
as it currently selects the `id` and `idsToSelect` on line 32 and 33?

Reply:
You could users our `unique` helper here, from `helpers/arrays`.